### PR TITLE
various fixes

### PR DESCRIPTION
**Pull request by petersteneteg**
Various minor fixes and updates
 * use /permissive- on VS17
 * use latest pybind
 * enable reorder warning on VS17
 * avoid including windows.h in the PCH to avoid some bugs
 * minor warning fixes

### CODE REVIEW

File: cmake/globalmacros.cmake [hunk 1]
```diff
@@ -610,35 +610,80 @@ endfunction()
...
 # Optimize compilation with pre-compilied headers from inviwo core
 function(ivw_compile_optimize_inviwo_core_on_target target)
-    if(PRECOMPILED_HEADERS)
-        ivw_cotire_ignore_on_target(${target})
-        set_target_properties(${target} PROPERTIES COTIRE_ADD_UNITY_BUILD FALSE)
-        get_target_property(_prefixHeader inviwo-core COTIRE_CXX_PREFIX_HEADER)
-        set_target_properties(${target} PROPERTIES COTIRE_CXX_PREFIX_HEADER_INIT "${_prefixHeader}")
-        cotire(${target})
-    endif()
+    message(DEPRECATION "Use ivw_compile_optimize_on_target")
```
Comment (martinfalk) [inline cmake/globalmacros.cmake:665]: how will this effect an existing CMake setup? Must the user change anything in order to enable the optimizations?

File: cmake/globalmacros.cmake [hunk 2]
```diff
@@ -610,35 +610,80 @@ endfunction()
...
+    set(tmp2 "")
+    while(NOT "${tmp1}" STREQUAL "${tmp2}")
+        set(tmp2 ${tmp1})
+        get_filename_component(tmp1 ${tmp1} DIRECTORY)
+    endwhile()
+    set(${retval} ${tmp1} PARENT_SCOPE)
 endfunction()
 
+if(WIN32 AND MSVC)
+    ivw_get_header_path("windows.h" ivw_private_windows_path)
```
Comment (martinfalk) [inline cmake/globalmacros.cmake:660]: when is this executed? Problem with race condition?

File: cmake/globalmacros.cmake [hunk 3]
```diff
@@ -610,35 +610,80 @@ endfunction()
...
     if(PRECOMPILED_HEADERS)
-        ivw_cotire_ignore_on_target(${target})
+        get_target_property(ignorePaths ${target} COTIRE_PREFIX_HEADER_IGNORE_PATH)
+        if(NOT ignorePaths)
+            set(ignorePaths "${CMAKE_CURRENT_SOURCE_DIR}")
+        endif()
+
+        list(APPEND ignorePaths "${IVW_EXTENSIONS_DIR}/warn")
+        if(WIN32 AND MSVC)
+            ivw_get_drive(${ivw_private_windows_path} windrive)
```
Comment (martinfalk) [inline cmake/globalmacros.cmake:679]: is `ivw_private_windows_path` already defined here? (race condition?)

**Review by martinfalk — APPROVED** [review]
see comments

File: cmake/globalmacros.cmake (hunk 1 above)
Comment (petersteneteg) [reply cmake/globalmacros.cmake:665]: No, I just forward the call to that one. I dont know why we used this one. Seems to come from some bad ideas about how pch works...

File: cmake/globalmacros.cmake (hunk 2 above)
Comment (petersteneteg) [reply cmake/globalmacros.cmake:660]: This should be executed when the file is first parsed, so to my understanding it should not be an issue...

File: cmake/globalmacros.cmake (hunk 3 above)
Comment (petersteneteg) [reply cmake/globalmacros.cmake:679]: From my understanding yes...

File: cmake/globalconfig.cmake [hunk 4]
```diff
@@ -276,13 +276,15 @@ option(IVW_RUNTIME_MODULE_LOADING
 
 #--------------------------------------------------------------------
 # Check if OpenMP is available and set it to use, and include the dll in packs
+option(OpenMP_ON "Use OpenMP" OFF)
```
Comment (ResearchDaniel) [inline cmake/globalconfig.cmake]: ON by default if found. Will make it work as before but still show the option when not found

File: include/inviwo/core/util/stdextensions.h [hunk 5]
```diff
@@ -208,7 +209,8 @@ template <class T, class... Types>
 struct for_each_type<std::tuple<T, Types...>> {
     template <class F, class... Args>
     auto operator()(F&& f, Args&&... args) {
-#ifdef _WIN32  // TODO: remove win fix when VS does the right thing...
+// TODO: remove win fix when VS does the right thing...
```
Comment (ResearchDaniel) [inline include/inviwo/core/util/stdextensions.h]: TODO can be removed? Just add explanation in the #ifdef instead

**Review by ResearchDaniel — APPROVED** [review]
Just some minor comments.

File: CMakeLists.txt [hunk 6]
```diff
@@ -228,3 +228,6 @@ include(cmake/packaging.cmake)
 # Generate Doxygen setup. 
 include(tools/doxygen/doxygen.cmake)
 make_doxygen_target(all_modules)
+
+string(TIMESTAMP time "%Y-%m-%d %H:%M:%S")
+message(STATUS "Configure ended at ${time}")
```
Comment (ResearchDaniel) [inline CMakeLists.txt:233]: Remove debug message?

File: cmake/taget_info.cmake [hunk 7]
```diff
@@ -0,0 +1,389 @@
...
+    get_property(s TARGET ${tgt} PROPERTY ${prop} SET)
+    # only produce output for values that are set
+    if(s)
+        message("Property = '${prop}' Defined='${d}' Set='${s}'")
+        message("   Value = '${v}'")
+    endif()
+endfunction()
+
+function(echo_target tgt)
+    set(props
```
Comment (ResearchDaniel) [inline cmake/taget_info.cmake]: Is this list of properties taken from somewhere? Could be good to add a comment in that case for possible future modifications

**Review by ResearchDaniel — APPROVED** [review]
Looks good. Some minor comments

File: CMakeLists.txt (hunk 6 above)
Comment (petersteneteg) [reply CMakeLists.txt:233]: No it not a debug message, I put it there so you can easily see when you last ran cmake so you can see if you should rerun it...

File: cmake/taget_info.cmake (hunk 7 above)
Comment (petersteneteg) [reply cmake/taget_info.cmake]: It’s from the docs... but yes...

**Pull request merged**